systems/robotInterfaces/+footstepPlanner/nonlinearCollocation.m

```diff
@@ -295,14 +295,15 @@
   c = zeros(nvar, 1);
 
   w_goal = diag(weights.goal);
-  if seed_plan.footsteps(end).frame_id == biped.foot_frame_id.right
-    xg = reshape(goal_pos.right, [], 1);
-  else
-    xg = reshape(goal_pos.left, [], 1);
+  for j = (nsteps - 1):nsteps
```

Comment:
maybe it would be better to use `max(1,nsteps-1)`, just in case you're doing a one-footstep plan?